sound/soc/sof/intel/hda-dsp.c

```diff
@@ -301,6 +301,14 @@ static int hda_suspend(struct snd_sof_dev *sdev, int state)
 	snd_sof_pci_update_bits(sdev, PCI_PGCTL,
 				PCI_PGCTL_LSRMD_MASK, PCI_PGCTL_LSRMD_MASK);
 
+	/* reset controller */
+	ret = hda_dsp_ctrl_link_reset(sdev, true);
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: failed to reset controller during suspend\n");
+		return ret;
+	}
+
 	return 0;
 }
 
```

Comment:
so we put/keep controller at reset(0) during suspend, do we need symmetric put/let controller out of reset(1) at resume?
I can imagine that we will do 0->1 in hda_dsp_probe(), but that probe() is not invoked at resume actually?

Reply:
@keyonjie the call to snd_hdac_bus_init_chip takes care of it but you are right for non-hd audio platforms we need to use the other call. let me fix it.

Reply:
fixed!